modules/contact/src/constraints/ADMortarLagrangeConstraint.C

```diff
@@ -159,7 +159,10 @@ ADMortarLagrangeConstraint::computeJacobian(Moose::MortarType mortar_type)
       _i = index;
       residuals_lower[index_lower] += _JxW_msm[_qp] * _coord[_qp] * computeQpResidual(mortar_type);
 
-      // Get rid of derivatives that we assume won't count (tolerance prescribed by user)
+      // TODO: Get rid of derivatives that we assume won't count (tolerance prescribed by user)
+      // This can cause zero diagonal terms with the variable condensation preconditioner when the
+      // no adaptivity option is used (dofs are not checked).
+
```

Comment:
This is unfortunate. Would it make sense to make a boolean user parameter that defaults to true to allow trimming, and that should be turned to false when running the VCP? Is the VCP setup through the `ContactAction` or no?

Reply:
Actually better yet would be during `initialSetup` to just check (`dynamic_cast`) the preconditioner type and don't trip if it's VCP. Then no additional user parameter required

Reply:
The VCP cannot be set up via contact action yet. It is still undergoing some work to make it more general for 3D applications/non-trivial geometries.

I can implement that strategy and set that flag to false for that particular test. Since we are removing Jacobian entries I am a bit hesitant to do this now should this have unintended consequences that we are not even catching in MOOSE testing. Anyway, I can give it a try and monitor it. The good part of this is that only the "no adaptivity" test is affected, which wouldn't be the most general case for VCP.   

But I am also getting a type mismatch when calling that routine from MOOSE: https://github.com/idaholab/moose/pull/20665#issuecomment-1085001008

Reply:
Can you paste the compiler error?

Reply:
https://civet.inl.gov/job/1010054/